### PR TITLE
Expose block censorship timeout in config

### PR DESCRIPTION
**Pull request by MayRosenbaum**
#### Type of change

- Improvement (improvement to code)
- Test update

#### Description

As described in issue #4401 

#### Related issues

issue #4401

### CODE REVIEW

File: core/deliverservice/config.go [hunk 1]
```diff
@@ -124,6 +130,24 @@ func (c *DeliverServiceConfig) loadDeliverServiceConfig() {
 	}
 	c.BlockGossipEnabled = enabledConfigOptionMissing || viper.GetBool(enabledKey)
 
+	blockCensorshipTimeout := "peer.deliveryclient.blockCensorshipTimeout"
+	blockCensorshipTimeoutOptionMissing := !viper.IsSet(blockCensorshipTimeout)
+	if blockCensorshipTimeoutOptionMissing {
+		c.BlockCensorshipTimeout = DefaultBlockCensorshipTimeout
+		logger.Infof("peer.deliveryclient.blockCensorshipTimeout is not set, defaulting to 30s.")
```
Comment (C0rWin) [inline core/deliverservice/config.go]:
```suggestion
		logger.Infof("peer.deliveryclient.blockCensorshipTimeout is not set, defaulting to %d.", DefaultBlockCensorshipTimeout)
```

File: core/deliverservice/config.go [hunk 2]
```diff
@@ -124,6 +130,24 @@ func (c *DeliverServiceConfig) loadDeliverServiceConfig() {
...
+	} else {
+		c.BlockCensorshipTimeout = viper.GetDuration(blockCensorshipTimeout)
+	}
+
+	minimalReconnectInterval := "peer.deliveryclient.minimalReconnectInterval"
+	MinimalReconnectIntervalOptionMissing := !viper.IsSet(minimalReconnectInterval)
+	if MinimalReconnectIntervalOptionMissing {
+		c.MinimalReconnectInterval = DefaultMinimalReconnectInterval
+		logger.Infof("peer.deliveryclient.minimalReconnectInterval is not set, defaulting to 100ms.")
```
Comment (C0rWin) [inline core/deliverservice/config.go]:
```suggestion
		logger.Infof("peer.deliveryclient.minimalReconnectInterval is not set, defaulting to %d ms.", DefaultMinimalReconnectInterval)
```

File: core/deliverservice/config.go (hunk 1 above)
Comment (C0rWin) [inline core/deliverservice/config.go]: you have

```
   viper.SetDefault(blockCensorshipTimeout, DefaultBlockCensorshipTimeout)
```

and there is no need to check whether the value was set or not.

**Review by C0rWin — DISMISSED** [review]

File: core/deliverservice/config.go (hunk 1 above)
Comment (MayRosenbaum) [reply core/deliverservice/config.go]: @C0rWin 
So you suggest replacing the code (lines 133-140) by the following code ? 

	blockCensorshipTimeout := "peer.deliveryclient.blockCensorshipTimeout"
	viper.SetDefault(blockCensorshipTimeout, DefaultBlockCensorshipTimeout)
	c.BlockCensorshipTimeout = viper.GetDuration(blockCensorshipTimeout)

File: core/deliverservice/config.go [hunk 3]
```diff
@@ -124,6 +130,24 @@ func (c *DeliverServiceConfig) loadDeliverServiceConfig() {
 	}
 	c.BlockGossipEnabled = enabledConfigOptionMissing || viper.GetBool(enabledKey)
 
+	blockCensorshipTimeout := "peer.deliveryclient.blockCensorshipTimeout"
```
Comment (tock-ibm) [inline core/deliverservice/config.go]: blockCensorshipTimeout => blockCensorshipTimeoutKey

File: core/deliverservice/config.go [hunk 4]
```diff
@@ -124,6 +130,24 @@ func (c *DeliverServiceConfig) loadDeliverServiceConfig() {
...
+	blockCensorshipTimeoutOptionMissing := !viper.IsSet(blockCensorshipTimeout)
+	if blockCensorshipTimeoutOptionMissing {
```
Comment (tock-ibm) [inline core/deliverservice/config.go:135]: is there a unit test that covers the case where the new config keys are missing, and check that the default is applied?

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**